pilot/cmd/pilot-agent/main.go

```diff
@@ -221,7 +221,7 @@ func initStatusServer(ctx context.Context, proxy *model.Proxy, proxyConfig *mesh
 
 func initStsServer(proxy *model.Proxy, tokenManager security.TokenManager) (*stsserver.Server, error) {
 	localHostAddr := localHostIPv4
-	if network.IsIPv6Proxy(proxy.IPAddresses) {
+	if proxy.IsIPv6() {
```

Comment:
In dual stack mode, it will always listen to ipv6 address? If so, I do not  think it is right.